File: genium/src/test/resources/smoke/packages/output/android/com/example/smoke/off/NestedPackages.java
/*
 *

 */
package com.example.smoke.off;
import android.support.annotation.NonNull;
import com.example.NativeBase;
public final class NestedPackages extends NativeBase {
    public final static class SomeStruct {
        @NonNull
        public String someField;
        public SomeStruct(@NonNull final String someField) {
            this.someField = someField;
        }
    }
    /**
     * For internal use only.
     * @exclude
     */
    protected NestedPackages(final long nativeHandle) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public static native NestedPackages.SomeStruct basicMethod(@NonNull final NestedPackages.SomeStruct input);
}

File: gluecodium/src/test/resources/smoke/packages/output/android-kotlin/com/example/smoke/off/NestedPackages.kt
/*

 *
 */

@file:JvmName("NestedPackagesExtensions")

package com.example.smoke.off

import com.example.NativeBase

class NestedPackages : NativeBase {

    class SomeStruct {
        @JvmField var someField: String



        constructor(someField: String) {
            this.someField = someField
        }





    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}







    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)


        @JvmStatic external fun basicMethod(input: NestedPackages.SomeStruct) : NestedPackages.SomeStruct
    }
}

